crates/nu-parser/src/parse_keywords.rs

```diff
@@ -170,7 +180,7 @@ pub fn parse_def_predecl(working_set: &mut StateWorkingSet, spans: &[Span]) {
                 working_set.error(ParseError::DuplicateCommandDef(spans[1]));
             }
         }
-    } else if name == b"extern" && spans.len() >= 3 {
+    } else if decl_name == b"extern" && spans.len() == 3 {
```

Comment:
```suggestion
    } else if decl_name == b"extern" && spans.len() >= 3 {
```

Reply:
I committed that change. let's see. thanks @pingiun !!